osquery/tables/system/linux/mounts.cpp

```diff
@@ -25,10 +25,18 @@ QueryData genMounts(QueryContext& context) {
     return {};
   }
 
+  std::vector<std::string> mnt_type_blacklist = {
+    "autofs"
+  };
+
   struct mntent* ent = nullptr;
   while ((ent = getmntent(mounts))) {
     Row r;
 
+    if (std::find(mnt_type_blacklist.begin(), mnt_type_blacklist.end(), std::string(ent->mnt_type)) != mnt_type_blacklist.end()) {
```

Comment:
The suggestion of using a `vector<const char*>` assumes that `ent->mnt_type` is a `const char*` and avoids having to copy the data into a `std::string`.

We'll also need to format this code. I'm hoping the CI tests will provide a diff for you. :)

Reply:
Thank you for the direction. I see the diff and will get the formatting cleaned up.

I apologize, as C++ is not my primary language, but if we don't copy `ent->mnt_type` into `std::string`, then doesn't that break the `find()` entirely? (I think that it will never match because its now trying to match pointers and not strings.) So then the correct way forward would be a good ol' loop over an array with a `strcmp()`, and no need for a vector or find at all?

Reply:
Done, but in a more C style way with a `strcmp()` check in a loop over the blacklist array. I'm fine with either way, but couldn't figure out how to get the `find()` to work without copying `ent->mnt_type` into a `std::string` first. Either method works as far as I'm concerned. Let me know what you think.